### PR TITLE
Products Template Improvization

### LINKED ISSUE #58: Products logic improvization

### PR DESCRIPTION
**Pull request by meejain**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #58 

Test URLs:
- Before: https://main--octoral--aemsites.hlx.live/en/products/non-voc/mixing_colour_system/octobase_system_mixing_colours
- After: https://issue58-products--octoral--aemsites.hlx.live/en/products/non-voc/mixing_colour_system/octobase_system_mixing_colours

### CODE REVIEW

File: scripts/utils.js [hunk 1]
```diff
@@ -12,3 +12,8 @@ export function getPathSegments() {
 export function normalizeString(str) {
   return str.toLowerCase().replace(/ /g, '_');
 }
+
+export function normalizeImage(str) {
+  const imagePath = 'https://main--octoral--aemsites.hlx.page/products/assets/';
```
Comment (dave-fink) [inline scripts/utils.js]: Why not just do something like this in products.js?

const productImage = `/product/assets/${normalizeString(entry.image)}`

Comment (meejain) [reply scripts/utils.js]: Here normalizeString has got a different return formula - > replace(/ /g, '_')

And we need - > replace(/_/g, '-')

Plus at the same time we would like to keep the location assignment of the images at one location via leveraging normalizeImage function. Apart from entry.image, we do have entry['type-image'] & entry['title-image'] as well. Hence its better to pass all of these strings into a function rather & getting us the required URL for the images.

Comment (akasjain2) [reply scripts/utils.js]: I think @dave-fink is suggesting using relative image link, without specifying the domain. 

Also, we should sanitize image path per EDS guidelines similar to https://github.com/adobe/helix-importer/blob/main/src/utils/FileUtils.js#L23 & https://github.com/adobe/helix-importer/blob/main/src/utils/FileUtils.js#L14.

Something like
```
export function normalizeImage(str) {
  const imagePath = '/products/assets/';
  return sanitizePath(imagePath + str);
}
```

Comment (meejain) [reply scripts/utils.js]: Sure. Leveraging relative links now 

export function normalizeImage(str) {
  const imagePath = '/products/assets/';
  return imagePath + str.toLowerCase().replace(/_/g, '-');
}

File: scripts/utils.js [hunk 2]
```diff
@@ -12,3 +12,8 @@ export function getPathSegments() {
 export function normalizeString(str) {
   return str.toLowerCase().replace(/ /g, '_');
 }
+
```
Comment (dave-fink) [inline scripts/utils.js]: Looks good!  I see the string replace is different - I missed that before.  My only comment is, would this function ever be used outside of products.js?  If yes, lets keep it here.  If not, then lets keep it together with products.js.

**Review by dave-fink — APPROVED** [review]

**Pull request merged**